src/vm/codeman.cpp

```diff
@@ -1427,6 +1441,37 @@ void EEJitManager::SetCpuInfo()
                 CPUCompileFlags.Clear(CORJIT_FLAGS::CORJIT_FLAG_USE_AVX2);
             }
         }
+
+        if (maxCpuId >= 0x07)
+        {
+            (void)getextcpuid(0, 0x07, buffer);
+
+            if ((buffer[4] & 0x08) != 0)            // BMI1
+            {
+                CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_BMI1);
+            }
+
+            if ((buffer[5] & 0x01) != 0)            // BMI2
```

Comment:
Shall we move the `BMI1/2` checks into the `Avx` block to make the dependency?

Reply:
We shouldn't place artificial dependencies. 

From what the CPU manuals (and various online sources) indicate, BMI1/BMI2 dont depend on AVX.

If, after you get the hardware to validate, it shows that they do require AVX, we should update this again at that time

Reply:
> after you get the hardware to validate, it shows that they do require AVX, we should update this again at that time

Probably, I can get the answer by the end of today.